org.jacoco.agent.rt.test/src/org/jacoco/agent/rt/internal/AgentModulTest.java

```diff
@@ -0,0 +1,92 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2023 Mountainminds GmbH & Co. KG and Contributors
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Evgeny Mandrikov - initial API and implementation
+ *
+ *******************************************************************************/
+package org.jacoco.agent.rt.internal;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotSame;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertTrue;
+
+import org.jacoco.core.test.validation.JavaVersion;
+import org.junit.AssumptionViolatedException;
+import org.junit.Test;
+
+/**
+ * Unit tests for {@link AgentModule}.
+ */
+public class AgentModulTest {
+
+	@Test
+	public void isSupported_should_return_false_before_Java9() {
+		if (JavaVersion.current().isAtLeast("9")) {
+			throw new AssumptionViolatedException(
+					"Modules available after Java 9");
+		}
+		assertFalse(AgentModule.isSupported());
+	}
+
+	@Test
+	public void isSupported_should_return_true_for_Java9_or_greater() {
+		if (JavaVersion.current().isBefore("9")) {
+			throw new AssumptionViolatedException(
+					"Modules only available after Java 9");
+		}
+		assertTrue(AgentModule.isSupported());
+	}
```

Comment:
Maybe simply

```
	@Test
	public void isSupported_should_return_false_before_Java9() {
		assertEquals(!JavaVersion.current().isBefore("9"),
				AgentModule.isSupported());
	}
```

to avoid marking tests as unsupported?